### PR TITLE
CO-579 fix docs freshness owner authority

### PR DESCRIPTION
**Pull request by Kbediako**
## Summary
- Re-home the live `docs:freshness:maintain` owner from terminal CO-575 to live CO-579 while preserving CO-568/CO-569 as exact baseline cohort owners.
- Add task-index-backed freshness authority for indexed task packet mirrors with explicit `task_index_review_overrides` audit output.
- Mirror exact rolling cohort owner resolution in `spec-guard`, add regressions, and record bounded pre-expiry spec review evidence.
- Link the broader lifecycle/finalizer root cause to CO-580 instead of widening this patch.

## Validation
- `node scripts/spec-guard.mjs --dry-run`
- `npm run docs:check`
- `MCP_RUNNER_TASK_ID=linear-5634bf98-fe59-41f5-8623-60c9ef0b2993 npm run docs:freshness`
- `node scripts/docs-freshness-maintain.mjs --check --format json`
- `npm run build`
- `npm run lint`
- `npm run test` (`367` files / `6267` tests passed in `/tmp/co579-test-final.log`)
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs` with explicit CO-579 scope override
- `npm run pack:smoke`
- `npm run review -- --uncommitted` final enforced review: `clean-success`, `review_verdict=clean`, valid clean contract, `finding_count=0`

## Notes
Diff budget override: CO-579 intentionally combines the docs-freshness authority fix, spec-guard owner-resolution rework, task-index override audit reporting, regression tests, task packet, registry/index owner updates, frontmatter review evidence, pack-smoke evidence, bounded pre-expiry spec review evidence, and admission recheck evidence required to keep spec-guard/docs-freshness truthful; splitting would separate the fix from the proof artifacts and risk reintroducing stale-owner drift.

- Shared-root control state after owner verification had `running=0`, `retrying=1`, and only CO-555 active-like in Rework; CO-527 is Backlog and not consuming WIP.

## Summary by CodeRabbit

* **Documentation**
  * Reassigned docs-freshness ownership to a live issue and updated guides, PRD, action plan and registry metadata.
  * Added a detailed implementation checklist, fallback decision table and handoff status.

* **New Features**
  * Freshness evaluation now honours newer task-index review dates and records review-source metadata.
  * Support for “retained terminal packet” evidence to prevent unintended archiving and new retained-item counts in reports.

* **Tests**
  * Expanded test coverage for cohort owner resolution, task-index overrides, retained-terminal handling and alias edge-cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Kbediako/CO/pull/881?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: scripts/docs-freshness.mjs [hunk 1]
```diff
@@ -218,6 +218,38 @@ function explicitTerminalTaskStatusLifecycle(taskStatus, entryPath) {
...
+    reviewSourceMetadata: {}
+  };
+  if (!entryPath || !registryReviewDate) {
+    return fallback;
+  }
+
+  const taskIndexEntry = lifecycleIndex.taskItemsByPath?.get(entryPath);
+  const taskIndexReviewDate = parseIsoDate(taskIndexEntry?.last_review);
+  if (!taskIndexEntry || !taskIndexReviewDate || taskIndexReviewDate <= registryReviewDate) {
+    return fallback;
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/docs-freshness.mjs:241]: Addressed in 80686f16ae: task-index review overrides are gated to active authoritative rows and unique path ownership, with regression coverage for stale rows.

File: scripts/lib/docs-freshness-lifecycle.js [hunk 2]
```diff
@@ -197,6 +214,10 @@ export function collectIndexedTaskPacketPaths(item, options = {}) {
     if (slugAlias) {
       addTaskPacketPathAliases(paths, slugAlias);
     }
+    const numericAlias = includeSlugAliases ? numberPrefixedTaskKeyAlias(taskKey, item) : null;
+    if (numericAlias) {
+      addTaskPacketPathAliases(paths, numericAlias);
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/lib/docs-freshness-lifecycle.js:241]: Addressed in 80686f16ae: numeric slug aliases now require unique alias ownership and cannot override colliding packet paths, with regression coverage.

File: scripts/lib/docs-freshness-lifecycle.js [hunk 3]
```diff
@@ -197,6 +236,10 @@ export function collectIndexedTaskPacketPaths(item, options = {}) {
...
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/lib/docs-freshness-lifecycle.js:241]: Addressed in 80686f16ae: terminal lifecycle evidence uses the same unambiguous primary/alias ownership gate before byPath authority is granted.

File: scripts/lib/docs-freshness-lifecycle.js [hunk 4]
```diff
@@ -221,20 +264,108 @@ function readSourceIssue(item) {
...
+    }
+    const primaryOwners = primaryOwnersByPath.get(docPath);
+    return !primaryOwners || (primaryOwners.size === 1 && primaryOwners.has(ownerKey));
+  }
+
+  for (const { item, taskKey, taskIndexEntry, ownerKey, primaryPaths, aliasPaths, allPaths } of indexedItems) {
+    if (taskIndexEntry.review_authoritative) {
+      for (const docPath of [
+        ...primaryPaths,
+        ...aliasPaths.filter((aliasPath) => canUseAliasPathForTaskIndex(aliasPath, ownerKey))
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/lib/docs-freshness-lifecycle.js:361]: Addressed in 80686f16ae: shared primary paths are rejected unless they have a single matching owner, so task-index review authority cannot be borrowed across tasks.
